src/core/featurehistory.cpp

```diff
@@ -72,6 +74,12 @@ void FeatureHistory::onBeforeCommitChanges()
     return;
   }
 
+  Tracker *layerTracker = mTrackingModel->trackerForLayer( vl );
```

Comment:
I think we should approach this slightly differently and connect to the layerInTrackingChanged( qgsvectorlayer *layer, bool tracking ) signal. When we get a layer tracking = true, we disconnect the layer signals (beforeCommitChanges,afterCommitChanges,committedFeaturesAdded,committedFeaturesRemoved, etc) and when the tracking is turned off, we reconnect.

Otherwise we'll still end up responding to quite a lot of signals. When debugging sensor stuff, where tracking can occur multiple times per second, I became quite aware of the cost of signals :)

Reply:
I considered this as premature optimization during development and also assumed C++ signals are quite cheap. Seems I have been wrong, so I accommodated the requested changes.

Reply:
Tested with TCP server tracking, seems it works as expected. 
- tracking enabled - no undo/redo, even for manually digitezed features
- tracking disabled - all good